python/pyspark/sql/connect/udf.py

```diff
@@ -253,15 +253,17 @@ def register(
             f = cast("UserDefinedFunctionLike", f)
```

Comment:
Before the change proposed in that PR, registering Arrow-optimized Python UDF is buggy: the UDF registered will be executed as a non-Arrow-optimized Python UDF.
That should be fixed separately.